### ISSUE TITLE
Error output message issues

### ISSUE THREAD

**Issue by fmatthew5876** [EasyRPG/Player#2075]
When we do `Output::Error` we turn the screen red and display a message. We have a wierd spin loop here and we don't redraw the screen. If you move a window over the player window, the error message won't be redrawn.

- [ ] Integrate`Output::Error` handling into the main loop.
- [ ] Redraw the error screen each frame.

**Comment by Ghabry** [comment]
> IntegrateOutput::Error handling into the main loop.

You are the (async) expert here: Do you consider this feasible?
Because we have no exception support on some platforms returning to the mainloop is a bit tricky.

**Comment by fmatthew5876** [comment]
Yes, without exceptions doing this in the main loop is impossible. Maybe some `setjmp` / `lomgjmp` hack is possible, but that won't unwind the stack and will cause corruption and crashes easily.

**Comment by CherryDT** [comment]
Can't you draw the screen again from the inner code maybe? That's what RM does. (Those things like transitions and battle waits which update the screen without advancing the frame counter, they are screen redraws and frame waits being called from inside the main loop's update handler.)

**Comment by fmatthew5876** [comment]
We can fix the redrawing issue that way yes, but I wonder what this means for libretro since it wants to own the main loop and we aren't ever returning to it.
